compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1012,6 +1013,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
                 let async_body = this.make_async_expr(
```

Comment:
Async closures should probably also support `#[track_caller]`

Reply:
It currently doesn't, so that's why I haven't changed it in this PR.

It isn't difficult change to make, but it does require lifting the attribute lower up from after `ExprKind` lowering to before it. Do you happen to know why we currently lower ID and attributes after `ExprKind`?

Reply:
Added. I then changed `Option<HirId>` to `HirId` to simplify code. We don't need to special case async blocks as `#[track_caller]` is not legal on async block. But I added a test anyway to avoid accidentally allow this on normal async block.

Reply:
Given the incremental issues, maybe we can revert the last change and go with `Option<HirId>`? Or are those test failures unrelated to the last change?